### DISCUSSION TITLE
Error while trying to create a dependency on github 3rd party package

### DISCUSSION THREAD

**Discussion by HuangLED** [xmake-io/xmake#4004] [General]
I defined like this: 

> add_repositories("libcommon git@github.com:HuangLED/libcommon.git")
> 
> add_requires("libcommon")
> 
> target("gstore")
>     set_kind("binary")
>     add_packages("libcommon")
> 

Then got 

> note: the following packages were not found in any repository (check if they are spelled correctly):
>   -> libcommon

I checked .xmake folder, the library is indeeded pulled to my local disk. 

What did I do wrong?

**Comment by waruqi** [comment]
> add_repositories("libcommon [git@github.com](mailto:git@github.com):HuangLED/libcommon.git")

add_repositories 是用来添加仓库的，一个仓库包含 N 个包定义，不是用来添加单个包的

类似这种，叫仓库。。https://github.com/xmake-io/xmake-repo

具体接口说明，仔细看下文档

**Reply by HuangLED** [reply]
懂了。

我确实仔细阅读了“[Using self-built private package repository]“ 这个章节，但是这个章节里面只讲了repository，但是没有提到有了repository之后，package跟它是什么关系，所以以为是一回事情。

那我再翻翻
